apiutils/task.go

```diff
@@ -0,0 +1,21 @@
+package apiutils
```

Comment:
The types are defined in `api`. We need a place to put utility functions that operate on these types, such as comparators. Normally these would go in `api` itself, but I understand that we want to reserve `api` for autogenerated code only.

I don't really care what the package is called and I'm happy to accept any other name. But I think we do need to have such a package. This question has come up several times, and copying code every time we encounter something that needs to be shared between the agent and the manager is not sustainable.

Or if we're okay with putting such non-autogenerated functions in `api`, that's fine with me too.